build/release/release.sh

```diff
@@ -52,6 +52,20 @@ if [[ ${RELEASE_VERSION} =~ .*-SNAPSHOT ]]; then
   exit 1
 fi
 
+if [[ -z ${JAVA_HOME} ]]; then
+  if [[ $(command -v java) ]]; then
+    # shellcheck disable=SC2046
+    export JAVA_HOME="$(dirname $(dirname $(which java)))"
```

Comment:
This silently affects all following procedures, we only need to find the right `java`, but won't want to set `JAVA_HOME`.

Reply:
assume there is a `java` at `/bin/java`, then the evaluated `JAVA_HOME` is `/`, does it make sense?

Reply:
How about just getting javacmd?
```
if [ -z "$JAVA_HOME" ] ; then
  JAVA_CMD="`\\unset -f command; \\command -v java`"
else
  JAVA_CMD="$JAVA_HOME/bin/java"
fi
```

Reply:
```
# Find the java binary
if [ -n "${JAVA_HOME}" ]; then
  JAVA="${JAVA_HOME}/bin/java"
else if [ "$(command -v java)" ]; then
  JAVA="java"
else
  echo "JAVA_HOME is not set" >&2
  exit 1
fi
```

Reply:
Thanks, fixed.